be/src/test/java/com/jjikmuk/sikdorak/integration/user/user/UserSearchIntegrationTest.java

```diff
@@ -0,0 +1,57 @@
+package com.jjikmuk.sikdorak.integration.user.user;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.jjikmuk.sikdorak.integration.InitIntegrationTest;
+import com.jjikmuk.sikdorak.user.user.controller.response.UserSimpleProfileResponse;
+import com.jjikmuk.sikdorak.user.user.service.UserService;
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@DisplayName("유저 검색 통합테스트")
+class UserSearchIntegrationTest extends InitIntegrationTest {
+
+    @Autowired
+    private UserService userService;
+
+    @Test
+    @DisplayName("유저의 닉네임으로 검색할 경우 유저 검색결과를 반환한다.")
+    void search_users_by_full_nickname() {
+        String searchNickname = testData.forky.getNickname();
+        List<UserSimpleProfileResponse> result = userService.searchUsersByNickname(searchNickname);
+
+        assertThat(result).isNotEmpty();
+        assertThat(result.get(0).nickname()).isEqualTo(testData.forky.getNickname());
+    }
+
+    @Test
+    @DisplayName("닉네임 일부분으로 검색할 경우 유저 검색결과를 반환한다.")
+    void search_users_by_part_nickname() {
+        String searchNickname = testData.forky.getNickname().substring(0, 1);
+        List<UserSimpleProfileResponse> result = userService.searchUsersByNickname(searchNickname);
+
+        assertThat(result).isNotEmpty();
+        assertThat(result.get(0).nickname()).isEqualTo(testData.forky.getNickname());
```

Comment:
get(0)으로 검색하면 테스트 데이터 추가될 때 0보다 먼저 나오는 경우 테스트가 깨질 거 같아요~

p3; list 돌면서 확인하는건 어떠세요?

```suggestion
assertThat(result)
    .extracting(result::getNickname)
    .contains("searchNickName");
```

ref : https://umanking.github.io/2021/06/26/assertj-iteration/

Reply:
오 좋은데요..? 바로 적용할게요!

Reply:
extracting 이런기능도 있었군요!
좋네요 ㅎㅎ